frb_codegen/src/library/codegen/ir/mir/ty/future.rs

```diff
@@ -0,0 +1,59 @@
+use crate::codegen::ir::mir::ty::{MirContext, MirType, MirTypeTrait};
+use crate::codegen::ir::mir::ty::boxed::MirTypeBoxed;
+use crate::codegen::ir::mir::ty::MirType::Boxed;
+
+crate::mir! {
+pub struct MirTypeFuture {
```

Comment:
I guess maybe we can simplify it a bit: As is mentioned in https://github.com/fzyzcjy/flutter_rust_bridge/issues/2172#issuecomment-2199517477, whenever we see `Pin<Box<dyn Future<Output = WhateverType>>>` (or sth else with futures), we *internally* treat it as if we are seeing a `async fn f() -> WhateverType`.

Then, we only need to change code in parser, and do not need to touch ir/generator.

Reply:
Yes indeed. The wire protocol generators are just stubs to make it compile.

For function return types, I do just use the Future Output type as the return, rather than the future type itself: https://github.com/fzyzcjy/flutter_rust_bridge/pull/2189/files#diff-c9fff61c90e7ff591fcdfa20d20d731a7f4f030f60d645ee3ee1e45d0c4212fcR42

Reply:
I see. Then if we never use the Future IR to generate anything and it is just a stub, then we may just write down empty code in places like https://github.com/fzyzcjy/flutter_rust_bridge/pull/2189/files#diff-7fa072c7ad1daba6b000e08671e3fbe0aed2ee654997cef838cc51ac500992e6R5.